file/writable_file_writer.cc

```diff
@@ -495,9 +495,6 @@ IOStatus WritableFileWriter::SyncWithoutFlush(const IOOptions& opts,
   IOStatus s = SyncInternal(io_options, use_fsync);
   TEST_SYNC_POINT("WritableFileWriter::SyncWithoutFlush:2");
   if (!s.ok()) {
-#ifndef NDEBUG
-    sync_without_flush_called_ = true;
```

Comment:
@anand1976  Can we remove sync_without_flush_called_? If we run  DBCompactionTest.ManualCompactionFailsInReadOnlyMode with this PR without this being removed, we can hit Assertion `sync_without_flush_called_' failed.

`mock_env->SetFilesystemActive(false);` will call set_seen_error(). Later, in DB close, we sync the WAL so will enter AssertFalseAndGetStatusForPrevError() while sync WAL has not been called. So it seems to me that 
"// This should only happen if SyncWithoutFlush() was called." is no longer true.

I also still don't understand why we want the below behavior in debug mode. Can we just "return IOStatus::IOError("Writer has previous error.");" instead of crashing?

```
  // SyncWithoutFlush() is the function that is allowed to be called
  // concurrently with other function. One of the concurrent call
  // could set seen_error_, and the other one would hit assertion
  // in debug mode.
 ```